### PR TITLE
Web

### PR DESCRIPTION
**Pull request by LeoBruant**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #(issue) | No issue linked.

## Type of change

Please add any relevant information to the type of change you've made.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update (or is a documentation update)
-   [ ] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

-   [x] Yes, I ran tests
-   [ ] No, I did not include any test
-   [ ] No test is needed for this change

### CODE REVIEW

File: api/internal/server/feature.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// @Failure 500  {object}  server.ErrorResponse
+// @Router /features [get]
+func (s *Server) GetFeatures(c *gin.Context) {
+	db := s.db.GetDB()
+
+	var features []model.Feature
+
+	if err := db.Find(&features).Error; err != nil {
+		s.InternalErrorResponse(c, err)
+
```
Comment (Spacelocust) [inline api/internal/server/feature.go:30]:
```suggestion
```

File: api/internal/server/routes.go [hunk 2]
```diff
@@ -12,6 +12,8 @@ import (
 func (s *Server) RegisterRoutes() http.Handler {
 	r := gin.Default()
 
+	r.Use(cors.Default())
```
Comment (Spacelocust) [inline api/internal/server/routes.go:17]:
```suggestion
        // Cors middleware
	r.Use(cors.Default())
```

File: api/internal/server/routes.go [hunk 3]
```diff
@@ -22,16 +24,18 @@ func (s *Server) RegisterRoutes() http.Handler {
...
 	s.RegisterMissionRoutes(r)
+	s.RegisterOauthRoutes(r)
 	s.RegisterObjectiveRoutes(r)
+	s.RegisterPlanetRoutes(r)
 	s.RegisterStratagemRoutes(r)
+	s.RegisterUsersRoutes(r)
 
 	// Create a new server for streaming planets and register the route for it
-	sse.NewServer(&s.db).PlanetsStream(r)
+	// sse.NewServer(&s.db).PlanetsStream(r)
```
Comment (Spacelocust) [inline api/internal/server/routes.go]:
```suggestion
	sse.NewServer(&s.db).PlanetsStream(r)
```

File: api/internal/server/user.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+// @Failure 500  {object}  server.ErrorResponse
+// @Router /users [get]
+func (s *Server) GetUsers(c *gin.Context) {
+	db := s.db.GetDB()
+
+	var users []model.User
+
+	if err := db.Find(&users).Error; err != nil {
+		s.InternalErrorResponse(c, err)
+
```
Comment (Spacelocust) [inline api/internal/server/user.go:30]:
```suggestion
```

File: web/lib/screens/login_screen.dart [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Form(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          TextFormField(
+            decoration: const InputDecoration(
+              labelText: 'Username',
```
Comment (YummYume) [inline web/lib/screens/login_screen.dart]: Traductions

**Review by YummYume — APPROVED** [review]

**Comment by YummYume** [comment]
:warning: attention la CI

File: api/migrations/20240709174100.sql
Comment (Spacelocust) [inline api/migrations/20240709174100.sql:1]: Il faudrait que tu supprimes les migrations que tu as faites pour enlever les migrations inutiles et faire une seule migration clean

**Comment by LeoBruant** [comment]
> ⚠️ attention la CI

nan jure

File: api/internal/server/feature.go (hunk 1 above)
Comment (LeoBruant) [reply api/internal/server/feature.go:30]: c'est moche

File: api/internal/server/user.go (hunk 4 above)
Comment (LeoBruant) [reply api/internal/server/user.go:30]: c'est moche

**Review by Spacelocust — APPROVED** [review]

**Pull request merged**